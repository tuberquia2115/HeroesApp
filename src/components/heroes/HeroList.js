import React, { useMemo } from 'react'
import { getHeroesByPublisher } from '../../selectors/getHeroesByPublisher'
import { HeroCard } from './HeroCard'


export const HeroList = ({ publisher }) => {
    let heroes = useMemo(() => getHeroesByPublisher(publisher), [publisher])
    return (
        <div className="row align-items-start animate__animated animate__fadeIn">
            {heroes.map((hero) => (
                <HeroCard key={hero.id} {...hero}/>
            ))}

        </div>

    )
}