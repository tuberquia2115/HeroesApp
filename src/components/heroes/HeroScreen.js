import React, { useMemo } from 'react'
import { Redirect, useParams } from 'react-router-dom'
import { getHeroById } from '../../selectors/getHeroById';
export const HeroScreen = ({ history }) => {
    const { heroId } = useParams();
    let hero = useMemo(() => getHeroById(heroId), [heroId])
    console.log(hero)
    if (!hero[0]) {
        return <Redirect to="/" />
    }

    const handlenReturn = () => {
        if (history.length <= 2) {
            history.push('/')
        } else {
            history.goBack()
        }
    }
    const {
        superhero,
        publisher,
        alter_ego,
        first_appearance,
        characters,
    } = hero[0];
    return (
        <div className="row mt-5">
            <hr />
            <div className="col-4">
                <img src={"https://img.pngio.com/iron-man-edwin-jarvis-hulk-falcon-marvel-heroes-2016-cc0-marvel-heroes-png-600_750.jpg"} className="img-thumbnail animate__animated animate__fadeInLeft" alt={superhero} />
            </div>
            <div className="col-8 animate__animated animate__fadeIn">
                <h3 className>{superhero}</h3>
                <ul className="list-group list-group-flush">
                    <li className="list-group-item"> <b> Alter ego: </b>{alter_ego} </li>
                    <li className="list-group-item"> <b> Publisher: </b>{publisher} </li>
                    <li className="list-group-item"> <b> First appearance: </b>{first_appearance} </li>
                </ul>
                <h5>Characters</h5>
                <p>{characters}</p>
                <button
                    onClick={handlenReturn}
                    className="btn btn-outline-info">
                    Return
                </button>
            </div>
        </div>
    )
}