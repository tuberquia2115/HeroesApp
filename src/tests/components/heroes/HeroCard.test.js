import React from 'react'
import { shallow } from 'enzyme'
import '@testing-library/jest-dom';
import { HeroCard } from '../../../components/heroes/HeroCard';

describe('Pruebas en <HeroCard/>', () => {

    const wrapper = shallow(
        <HeroCard
            id={'dc-batman'}
            superhero={'Batman'}
            alter_ego={'Bruce Wayne'}
            first_appearance={'Detective Comics #27'}
            characters={'Bruce Wayne'}
        />
    )
    test('debe mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot()
    })

})
