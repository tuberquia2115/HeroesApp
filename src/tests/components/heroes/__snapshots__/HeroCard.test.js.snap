// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HeroCard/> debe mostrarse correctamente 1`] = `
<div
  className="card ms-3"
  style={
    Object {
      "maxWidth": "540px",
    }
  }
>
  <div
    className="row no-gutters"
  >
    <div
      className="col-md-4"
    >
      <img
        alt="Batman"
        className="card-img"
        src="dc-batman.jpg"
      />
    </div>
    <div
      className="col-md-8 "
    >
      <div
        className="card-body"
      >
        <h5
          className="card-title"
        >
          Batman
        </h5>
        <p
          className="card-text"
        >
          Bruce Wayne
        </p>
        <p
          className="card-text"
        >
          <small
            className="text-muted"
          >
            Detective Comics #27
          </small>
        </p>
        <Link
          to="./hero/dc-batman"
        >
          M√ÅS ...
        </Link>
      </div>
    </div>
  </div>
</div>
`;
